src/filesystem-upload.js

```diff
@@ -57,17 +57,18 @@ export default class FileSystemUpload extends DirectBinaryUpload {
      *  passed in successful resolution will be an instance of UploadResult.
      */
     async upload(options, localPaths) {
-        const httpClient = new HttpClient(this.getOptions(), options);
-        const concurrentQueue = new ConcurrentQueue(this.getOptions(), options);
-        const partUploader = new PartUploader(this.getOptions(), options, httpClient, concurrentQueue);
-        const uploadResult = new UploadResult(this.getOptions(), options);
-        await this.createTargetFolder(options, httpClient);
+        const fileSystemUploadOptions = FileSystemUploadOptions.fromOptions(options);
+        const httpClient = new HttpClient(this.getOptions(), fileSystemUploadOptions);
```

Comment:
I don't expect the content of `this.getOptions()` to change during initialization, so maybe use a variable?

```suggestion
        const uploadOptions = this.getOptions(); 
        const httpClient = new HttpClient(this.getOptions(), fileSystemUploadOptions);
```